airflow/models/taskreschedule.py

```diff
@@ -94,18 +96,16 @@ def __init__(
         self.reschedule_date = reschedule_date
         self.duration = (self.end_date - self.start_date).total_seconds()
 
-    @staticmethod
-    @provide_session
+    @classmethod
     def query_for_task_instance(
+        cls,
```

Comment:
I hope this class-methods not part of public interface, and we can change signature for migrate out of legacy SQLAlchemy Query API